### PR TITLE
Added method pluck(property)

### LINKED ISSUE #11: New Method: pluck(property)

### PR DESCRIPTION
**Pull request by islas27**
- Added new method count(spec) to Array.prototype
- Added the unit test
- Added the documentation & examples
- Added new test data

Closes #11

### CODE REVIEW

File: aee.js [hunk 1]
```diff
@@ -72,3 +72,13 @@ Array.prototype.count = function (spec) {
   }
   return result
 }
+
+Array.prototype.pluck = function (property) {
+  if (property === undefined) return null
+  const len = this.length
+  let result = []
+  for (let i = 0; i < len; i += 1) {
+    result.push(this[i].hasOwnProperty(property) ? this[i][property] : null)
```
Comment (ryanez) [inline aee.js]: What will happen when the array contains `null` elements?

File: aee.js [hunk 2]
```diff
@@ -72,3 +72,13 @@ Array.prototype.count = function (spec) {
   }
   return result
 }
+
+Array.prototype.pluck = function (property) {
+  if (property === undefined) return null
```
Comment (ryanez) [inline aee.js]: `property` is a mandatory arguments, we would probably better throw an exception.

Comment (islas27) [reply aee.js]: fixed in b4b417c

File: aee.js (hunk 1 above)
Comment (islas27) [reply aee.js]: How about changing the conditional to `this[i[ && this[i].hasOwnProperty(property)` ? Added it in a1523dd

File: aee.js [hunk 3]
```diff
@@ -72,3 +72,13 @@ Array.prototype.count = function (spec) {
   }
   return result
 }
+
+Array.prototype.pluck = function (property) {
+  if (property === undefined) throw new ReferenceError('property is undefined')
+  const len = this.length
+  let result = []
+  for (let i = 0; i < len; i += 1) {
+    result.push(this[i] && this[i].hasOwnProperty(property) ? this[i][property] : null)
```
Comment (ryanez) [inline aee.js]: It's ok, it also could be
```js
let value = this[i];
result.push(value && value[property])
```

Comment (islas27) [reply aee.js]: Runned the tests, and it starts throwing `undefined` in objects where the property is not found. Should this be the correct behaviour? Right now it sends null both when an element is null, or if the object lacks the property sent to the function.

Comment (islas27) [reply aee.js]: screen shot 2016-12-27 at 3 50 44 pm

Comment (ryanez) [reply aee.js]: You decide what to return, but whatever your choice is don't forget to properly document it. Remember that if it's properly documented it's not a bug, it's a feature 😉

Comment (islas27) [reply aee.js]: Alright, I prefer to keep the already coded behavior, as a `null` value can be both understood as there was a missing object or the object did not contain the selected property while an `undefined` I think feels more like a failure in the function. Will add it to the documentation and merge it.

**Pull request merged**